### PR TITLE
[Taxonomy] Component docs update

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q             | A
| ------------- | ---
| Doc fix?      | yes
| New docs?     | yes
| Fixed tickets | 

The docs are a bit futuristic, because there is no ``new Taxonomy($taxon);`` in the current version of the component. :) But it will be done soon.

### CODE REVIEW

File: components/Taxonomy/interfaces.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+--------------
+
+
+The ``TaxonInterface`` gives an object an ability to have Taxons assigned as children.
+
+
+TaxonsAwareInterface
+--------------------
+
+The ``TaxonsAwareInterface`` should be implemented by models that support taxons.
```
Comment (pjedrzejewski) [inline components/Taxonomy/interfaces.rst]: ``that can be classified with taxons``

File: components/Taxonomy/models.rst [hunk 2]
```diff
@@ -4,116 +4,124 @@ Models
...
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxon-interface`.
+
+
+.. _component_taxonomy_model_taxon-translation:
+
+TaxonTranslation
+----------------
```
Comment (tuka217) [inline components/Taxonomy/models.rst]: I'm not sure, but maybe here should be some comment about how this class works?

File: components/Taxonomy/models.rst [hunk 3]
```diff
@@ -4,116 +4,124 @@ Models
...
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxonomy-interface`.
+
+
+.. _component_taxonomy_model_taxonomy-translation:
+
+TaxonomyTranslation
+-------------------
```
Comment (tuka217) [inline components/Taxonomy/models.rst]: Maybe here sould be an comment about how this class work.

File: components/Taxonomy/models.rst [hunk 4]
```diff
@@ -4,116 +4,124 @@ Models
...
-    |-- SuperTees
-    |-- Stickypicky
-    |-- Mugland
-    `-- Bookmania
+    |\__SuperTees
+    |\__Stickypicky
+    |\__Mugland
+     \__Bookmania
+
+
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: Extra blank line.

File: components/Taxonomy/models.rst [hunk 5]
```diff
@@ -4,116 +4,124 @@ Models
...
+|                 | and stored on ``TaxonomyTranslation``                           |
++-----------------+-----------------------------------------------------------------+
+| root            | First, root Taxon                                               |
++-----------------+-----------------------------------------------------------------
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxonomy-interface`.
+
+
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: Extra blank line.

File: components/Taxonomy/models.rst [hunk 6]
```diff
@@ -4,116 +4,124 @@ Models
...
-| removeTaxon(TaxonInterface $taxon) | Remove a taxon                  | TaxonomyInterface  |
-+------------------------------------+---------------------------------+--------------------+
\ No newline at end of file
++-----------------+-----------------------------------------------------------------+
+| Attribute       | Description                                                     |
++=================+=================================================================+
+| id              | Unique id of the taxonomy                                       |
++-----------------+-----------------------------------------------------------------+
+| name            | Name of the taxonomy taken from the root's ``TaxonTranslation`` |
++                 +                                                                 +
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: Strange structure, why we need + here?

File: components/Taxonomy/models.rst [hunk 7]
```diff
@@ -4,116 +4,124 @@ Models
...
++=================+=================================================================+
+| id              | Unique id of the taxonomy translation                           |
++-----------------+-----------------------------------------------------------------+
+| name            | Name of the taxonomy taken from the root's ``TaxonTranslation`` |
++-----------------+-----------------------------------------------------------------+
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxonomy-translation-interface`.
+
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: Extra blank line.

File: components/Taxonomy/models.rst [hunk 8]
```diff
@@ -4,116 +4,124 @@ Models
...
-+------------------------------------+---------------------------------+--------------------+
-| hasTaxon(TaxonInterface $taxon)    | Checks whether object has taxon | Boolean            |
-+------------------------------------+---------------------------------+--------------------+
-| addTaxon(TaxonInterface $taxon)    | Add a taxon                     | TaxonomyInterface  |
-+------------------------------------+---------------------------------+--------------------+
-| removeTaxon(TaxonInterface $taxon) | Remove a taxon                  | TaxonomyInterface  |
-+------------------------------------+---------------------------------+--------------------+
\ No newline at end of file
++-----------------+-----------------------------------------------------------------+
+| Attribute       | Description                                                     |
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: We should keep this consistent. I see @tuka217 is using Property - which seems right.

File: components/Taxonomy/models.rst [hunk 9]
```diff
@@ -4,116 +4,124 @@ Models
...
++-----------------+--------------------------------------------------------------------+
+| createdAt       | Date when taxon was created                                        |
++-----------------+--------------------------------------------------------------------+
+| updatedAt       | Date of last update                                                |
++-----------------+--------------------------------------------------------------------+
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxon-interface`.
+
```
Comment (pjedrzejewski) [inline components/Taxonomy/models.rst]: Extra blank line.

**Comment by pjedrzejewski** [comment]
Nice work! Let's do few tweaks and get it merged tomorrow (today).

File: components/Taxonomy/models.rst (hunk 6 above)
Comment (TheMadeleine) [inline components/Taxonomy/models.rst]: The description is sooo long, that I wanted to split it into two lines :D

File: components/Taxonomy/basic_usage.rst [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    use Sylius\Component\Taxonomy\Model\Taxonomy;
+
+    // Let's assume we want to begin creating new taxonomy in our system
+    // therefore we think of a new taxon that will be a root for us.
+    $taxon = new Taxon();
+
+    // And later on we create a taxonomy with our taxon as a root.
+    $taxonomy = new Taxonomy($taxon);
+
+    //before we can start using the newly created taxonomy, we have to define its locales.
```
Comment (NeverResponse) [inline components/Taxonomy/basic_usage.rst]: Space + capital = `// Before ...`

File: components/Taxonomy/interfaces.rst [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+Models Interfaces
+-----------------
+
+.. _component_taxonomy_model_taxonomy-interface:
+
+TaxonomyInterface
+~~~~~~~~~~~~~~~~~
+
+The ``TaxonomyInterface`` gives an object a possibility to manage Taxons.
```
Comment (NeverResponse) [inline components/Taxonomy/interfaces.rst]: Why `` instead of **?

File: components/Taxonomy/models.rst [hunk 12]
```diff
@@ -4,116 +4,124 @@ Models
...
++-----------------+--------------------------------------------------------------------+
+| name            | Name of the taxon taken form the ``TaxonTranslation``              |
++-----------------+--------------------------------------------------------------------+
+| slug            | Urlized name taken from the ``TaxonTranslation``                   |
++-----------------+--------------------------------------------------------------------+
+| permalink       | Full permalink for given taxon taken form the ``TaxonTranslation`` |
++-----------------+--------------------------------------------------------------------+
+| description     | Description of taxon taken from the ``TaxonTranslation``           |
++-----------------+--------------------------------------------------------------------+
+| taxonomy        | Taxonomy                                                           |
```
Comment (NeverResponse) [inline components/Taxonomy/models.rst]: taxonomy = Taxonomy
Yeah, I have no further questions :tada:

Comment (TheMadeleine) [inline components/Taxonomy/models.rst]: Captain Obvious always at your disposal @NeverResponse :D

File: components/Taxonomy/models.rst [hunk 13]
```diff
@@ -4,116 +4,122 @@ Models
...
+
+TaxonomyTranslation
+-------------------
+
++-----------------+-----------------------------------------------------------------+
+| Property        | Description                                                     |
++=================+=================================================================+
+| id              | Unique id of the taxonomy translation                           |
++-----------------+-----------------------------------------------------------------+
+| name            | Name of the taxonomy taken from the root's ``TaxonTranslation`` |
```
Comment (NeverResponse) [inline components/Taxonomy/models.rst]: Same deal as in interfaces, those `` thingies.

File: components/Taxonomy/models.rst [hunk 14]
```diff
@@ -4,116 +4,122 @@ Models
...
+| level           | How deep it is in the tree                                         |
++-----------------+--------------------------------------------------------------------+
+| createdAt       | Date when taxon was created                                        |
++-----------------+--------------------------------------------------------------------+
+| updatedAt       | Date of last update                                                |
++-----------------+--------------------------------------------------------------------+
+
+.. note::
+
+    This model implements the :ref:`component_taxonomy_model_taxon-interface`.
```
Comment (NeverResponse) [inline components/Taxonomy/models.rst]: Missing Sylius API links.

File: components/Taxonomy/models.rst (hunk 13 above)
Comment (TheMadeleine) [inline components/Taxonomy/models.rst]: here, i'd leave them. Because they really look better in the table :)

Comment (NeverResponse) [inline components/Taxonomy/models.rst]: :+1:

File: components/Taxonomy/models.rst (hunk 14 above)
Comment (TheMadeleine) [inline components/Taxonomy/models.rst]: added :)

File: components/Taxonomy/models.rst (hunk 3 above)
Comment (TheMadeleine) [inline components/Taxonomy/models.rst]: Added :)

File: components/Taxonomy/models.rst [hunk 15]
```diff
@@ -4,116 +4,134 @@ Models
...
+| Property        | Description                                                     |
++=================+=================================================================+
+| id              | Unique id of the taxonomy                                       |
++-----------------+-----------------------------------------------------------------+
+| name            | Name of the taxonomy taken from the root's ``TaxonTranslation`` |
++                 +                                                                 +
+|                 | and stored on ``TaxonomyTranslation``                           |
++-----------------+-----------------------------------------------------------------+
+| root            | First, root Taxon                                               |
++-----------------+-----------------------------------------------------------------
```
Comment (NeverResponse) [inline components/Taxonomy/models.rst]: Table doesn't render correctly, missing the ending `+`.

**Pull request merged**